appinsights/src/telemetry/remote_dependency.rs

```diff
@@ -112,6 +111,38 @@ impl RemoteDependencyTelemetry {
     pub fn measurements_mut(&mut self) -> &mut Measurements {
         &mut self.measurements
     }
+
+    /// Sets the dependency id. Use this to link other telemetry to this dependency by setting their operation
+    /// parent id to this id.
+    ///
+    /// ```rust,no_run
+    /// # use appinsights::TelemetryClient;
+    /// # use appinsights::telemetry::{RemoteDependencyTelemetry, SeverityLevel, Telemetry, TraceTelemetry};
+    /// # use http::{Method, Uri};
+    /// # use std::time::Duration;
+    /// # let client = TelemetryClient::new("<instrumentation key>".to_string());
+    /// let operation_id = "...".to_string();
+    /// let dependency_id = "...".to_string();
+    ///
+    /// let mut dependency = RemoteDependencyTelemetry::new(
+    ///     "Data processing",
+    ///     "InProc",
+    ///     Duration::from_millis(42),
+    ///     "internal",
+    ///     true,
+    /// );
+    /// dependency.set_id(dependency_id.clone());
+    /// dependency.tags_mut().operation_mut().set_id(operation_id.clone());
+    /// client.track(dependency);
+    ///
+    /// let mut trace = TraceTelemetry::new("Start", SeverityLevel::Information);
+    /// trace.tags_mut().operation_mut().set_id(operation_id);
+    /// trace.tags_mut().operation_mut().set_parent_id(dependency_id);
+    /// client.track(trace);
+    /// ```
+    pub fn set_id(&mut self, id: String) {
```

Comment:
Yes, you are right. I feel `impl Into<String>` usually provide better ergonomics